haystack/preview/components/retrievers/memory.py

```diff
@@ -39,28 +42,29 @@ class Output(ComponentOutput):
 
         documents: List[List[Document]]
 
-    def __init__(
-        self,
-        document_store_name: str,
-        filters: Optional[Dict[str, Any]] = None,
-        top_k: int = 10,
-        scale_score: bool = True,
-    ):
+    def __init__(self, filters: Optional[Dict[str, Any]] = None, top_k: int = 10, scale_score: bool = True):
         """
         Create a MemoryRetriever component.
 
-        :param document_store_name: The name of the MemoryDocumentStore to retrieve documents from.
         :param filters: A dictionary with filters to narrow down the search space (default is None).
         :param top_k: The maximum number of documents to retrieve (default is 10).
         :param scale_score: Whether to scale the BM25 score or not (default is True).
 
         :raises ValueError: If the specified top_k is not > 0.
         """
-        self.document_store_name = document_store_name
         if top_k <= 0:
             raise ValueError(f"top_k must be > 0, but got {top_k}")
         self.defaults = {"top_k": top_k, "scale_score": scale_score, "filters": filters or {}}
 
+    def warmup(self):
+        """
+        Double-checks that a store is present before running this component in a pipeline.
+        """
+        if not self.store:
+            raise ValueError(
+                "MemoryRetriever needs a store to run: use the 'store' parameter of 'add_component' to connect them."
+            )
```

Comment:
I would completely remove this and move made `add_component` stricter.

Reply:
Re-reading `add_component`, I realized it already checks for missing stores. We don't need this probably.